bin/shutter

```diff
@@ -5967,22 +5981,20 @@ sub STARTUP {
 		#fullscreen screenshot
 		if ($data eq "full" || $data eq "tray_full") {
 
-			$screenshooter = Shutter::Screenshot::Workspace->new(
-				$sc, $include_cursor, $delay_value,
-				$notify_timeout_active->get_active,
-				$wnck_screen ? $wnck_screen->get_active_workspace : undef,
-				undef, undef, $current_monitor_active->get_active
-			);
-
-			if ($ENV{XDG_SESSION_TYPE} eq "wayland") {
+			if ($x11_supported) {
+				$screenshooter = Shutter::Screenshot::Workspace->new(
+					$sc, $include_cursor, $delay_value,
+					$notify_timeout_active->get_active,
+					$wnck_screen ? $wnck_screen->get_active_workspace : undef,
+					undef, undef, $current_monitor_active->get_active
+				);
+				$screenshot = $screenshooter->workspace();
+			} else {
```

Comment:
Now this will trigger for `XDG_SESSION_TYPE="foo"`. Is that intended?

Reply:
It should trigger only on X11, not on Wayland. If we change the definition of x11_supported, it will trigger by default, unless it is explicitly Wayland. It fails on Wayland, so it's now a question of: Can XDG_SESSION_TYPE be not equal to "wayland" on a Wayland setup?

Reply:
Probably not. Unless for some reason user changed it himself

Reply:
I propose to redefine x11_supported the way you proposed, leave this check the way it is and see if issue reports will pop up, what do you think?

Reply:
Sure